src/Module/BaseApi.php

```diff
@@ -160,6 +175,55 @@ public static function getRequest(array $defaults)
 		return $request;
 	}
 
+	/**
+	 * Set boundaries for the "link" header
+	 * @param array $boundaries
+	 * @param int $id
+	 * @return array
+	 */
+	protected static function setBoundaries(int $id)
+	{
+		if (!isset(self::$boundaries['min'])) {
+			self::$boundaries['min'] = $id;
+		}
+
+		if (!isset(self::$boundaries['max'])) {
+			self::$boundaries['max'] = $id;
+		}
+
+		self::$boundaries['min'] = min(self::$boundaries['min'], $id);
+		self::$boundaries['max'] = max(self::$boundaries['max'], $id);
+	}
+
+	/**
+	 * Set the "link" header with "next" and "prev" links
+	 * @return void
+	 */
+	protected static function setLinkHeader()
+	{
+		if (empty(self::$boundaries)) {
+			return;
+		}
+
+		$request = self::$request;
+
+		unset($request['min_id']);
+		unset($request['max_id']);
+		unset($request['since_id']);
+
+		$prev_request = $next_request = $request;
+
+		$prev_request['min_id'] = self::$boundaries['max'] + 1;
```

Comment:
Have you verified no item is skipped with this +1? It wasn't part of my original implementation for the follow request list.

Reply:
The minimum and maximum values for the current request are defined in the `$boundaries` array. So the next or previous items have to be one item before or after, shouldn't they?

Reply:
How would you know if you skipped one item? Without knowing if the comparison is `id > min_id` or `id >= min_id` we have to test manually by altering the parameter ourselves.

Reply:
It looks like they are doing `id > min_id`, so the `+ 1` is unwarranted.

https://github.com/tootsuite/mastodon/blob/c3aef491d66aec743a3a53e934a494f653745b61/app/models/concerns/paginable.rb#L19
```suggestion
		$prev_request['min_id'] = self::$boundaries['max'];
```

Reply:
Details :grin: